tests/test_transformations/test_knn_clustering.py

```diff
@@ -0,0 +1,137 @@
+import unittest
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from sklearn.datasets import load_iris
+from spac.transformations import knn_clustering
+
+
+class TestKnnClustering(unittest.TestCase):
+    def setUp(self):
+        """
+        This method is run before each test.
+
+        It sets up a test AnnData object `syn_data` with the following attributes:
+
+        `syn_data.X`
+            A layer with 1000 rows and 2 columns generated from a normal distribution [500 rows ~ N(10, 1), 500 rows ~ N(100, 1)]
+
+        `syn_data.obs['classes']`
+            Class labels for rows: label -1 = missing, label 0 = mean 10, label 1 = mean 100
+            Approx 50% of the rows are missing label
+        """
+
+        n_rows = 1000
+
+        # Generate 1000 rows, half with mean (10, 10) and half with mean (100, 100)
+        mean_10 = np.random.normal(loc = 10, scale = 1, size=(n_rows // 2, 2))
+        mean_100 = np.random.normal(loc = 100, scale = 1, size=(n_rows // 2, 2))
+        data_rows = np.vstack((mean_10, mean_100))
+
+        # Generate class labels, label 0 = mean 10, label 1 = mean 100
+        class_labels = np.array([0] * (n_rows // 2) + [1] * (n_rows // 2), dtype=object)
+
+        # Replace ~50% of class labels with "missing" values
+        mask = np.random.rand(*class_labels.shape) < 0.5
+        class_labels[mask] = "no_label"
+
+        # Combine data columns with class labels
+        self.syn_dataset = data_rows
+
+        self.syn_data = AnnData(
+                self.syn_dataset,
+                var=pd.DataFrame(index=['gene1',
+                                        'gene2'])
+                )
+        self.syn_data.layers['counts'] = self.syn_dataset
+
+        self.syn_data.obsm["derived_features"] = \
+            self.syn_dataset
+
+        self.syn_data.obs['classes'] = class_labels
+        self.syn_features = ['gene1', 'gene2']
+
+        # The string for column where class labels stored in obs
+        self.label = "classes"
```

Comment:
let's change all instances of self.label to self.annotation. a similar comment was made in the knn_clustering function. we want to be consistent with the spac terminology.